File: 14-Longest_Common_Prefix.cpp
//https://leetcode.com/problems/longest-common-prefix/description/
class Solution {
public:
    string get_common(string str1, string str2){
        string res = "";
        for(int i=0; i<str1.size() && i < str2.size(); i++){
            if(str1[i] == str2[i]){
                res += str1[i];
            }else
                break;
        }
        return res;
    }
    string longestCommonPrefix(vector<string>& strs) {
        // sort(strs.begin(), strs.end());
        string res = strs[0];
        for(unsigned int i=1; i<strs.size(); i++){
            res = get_common(res, strs[i]);
            if(res.size() == 0)
                return "";
        }

        return res;
    }
};

// ["aca","cba"]
// ["rat", "a", "rattled"]
// ["a", "a", "b"]

File: 14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, words: List[str]) -> str:
        words.sort()
        index = 0
        for i_0 in range(len(words[0])):
            for i in range(1, len(words)):
                if(words[0][i_0] != words[i][i_0]):
                    return words[0][0:i_0]
        return words[0]
